docs/db/week-3/lesson.md

```diff
@@ -344,45 +337,35 @@ app.post("/customers", function (req, res) {
   const newName = req.body.name;
   const newEmail = req.body.email;
   const newPhone = req.body.phone;
-  // etcetera ...
-
-  //
-  // First validate the phone number using a regular expression to check for valid characters
-  //
-  if (newPhone.replace(/[+\-()0-9 ]/g, '0') ==    // replace all valid chars with 0
-      '0'.padEnd(newPhone.length, '0')) {
-    return res
-      .status(400)
-      .send("The phone number must only contain 0-9, +, -, (, ) or space.");
-  }
+  // et cetera ...
 
   //
   // Validate the new customer's email address by querying the existing customers table
   // to return any rows that contain the same values
   //
-  db.query("SELECT 1 FROM customers WHERE email=$1", [newEmail],
-            (err, result) => {
-      if (result.rowCount > 0) {      // note the use of result.rowCount
-        return res
-          .status(400)
-          .send("A customer with that email address already exists!");
-      } else {
-        const query =
-        "INSERT INTO customers (name, email, phone, address, city, postcode, country) " +
-          "VALUES ($1, $2, $3, $4, $5, $6, $7)";
-        db.query(query, [newName, newEmail, ..., newCountry],
-          (err) => {
-            res.send("Customer created.");
-          }
-      }
-    });
-});
+db.query("SELECT 1 FROM customers WHERE email=$1", [newEmail])
+  .then(function(result) {
+    if (result.rowCount > 0) {      // note the use of result.rowCount
+      return Promise.reject({error: 'Customer already exists'})
+    } else {
```

Comment:
This is a naive implementation. Better to add a unique constraint on the `email` column: however I left this in as it is a good opportunity for a refactor. Worthwhile to come up with an exercise on this.